kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -849,13 +851,62 @@ class SnapshotManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
 
   test("getLogSegmentForVersion: corrupt _last_checkpoint with empty delta log") {
     // listDeltaAndCheckpointFiles = Optional.empty()
-    val logSegmentOpt = snapshotManager.getLogSegmentForVersion(
+    val logSegmentOpt = snapshotManager.getLogSegmentAtOrBeforeVersion(
       createMockFSListFromEngine(Seq.empty),
       Optional.of(1),
+      Optional.empty(),
       Optional.empty()
     )
     assert(!logSegmentOpt.isPresent())
   }
+
+  /* ------------------- COORDINATED COMMITS TESTS ------------------ */
+  test("read with getCommits return empty lists") {
+    testWithCheckpoints(
+      (0L to 50L),
+      Seq(10, 30, 50),
+      Seq(20, 40),
+      numParts = 5,
+      tableCommitCoordinatorClientHandlerOpt =
+        Optional.of(new MockTableCommitCoordinatorClientHandler(logPath))
+    )
+  }
+
+  test("read with getCommits return a list with serveral unbackfilled commits") {
+    val unbackfilledCommits1 = Seq(51L)
+    testWithCheckpoints(
+      (0L to 50L),
+      Seq(10, 30, 50),
+      Seq(20, 40),
+      numParts = 5,
+      unbackfilledDeltaVersions = unbackfilledCommits1,
+      tableCommitCoordinatorClientHandlerOpt =
+        Optional.of(new MockTableCommitCoordinatorClientHandler(logPath, unbackfilledCommits1))
+    )
+
+    val unbackfilledCommits2 = 51L to 60L
+    testWithCheckpoints(
+      (0L to 50L),
+      Seq(10, 30, 50),
+      Seq(20, 40),
+      numParts = 5,
+      unbackfilledDeltaVersions = unbackfilledCommits2,
+      tableCommitCoordinatorClientHandlerOpt =
+        Optional.of(new MockTableCommitCoordinatorClientHandler(logPath, unbackfilledCommits2))
+    )
```

Comment:
what about the case where the backfilled and unbackfilled commits are overlapped?